### PR TITLE
Earnings chart updates

### PR DESCRIPTION
**Pull request by danielcolinjames**
updates on earnings charts:
- use midnight UTC for earnings buckets instead of rolling window based on current time
- update period options from [24h, 7d, 30d] to [7d, 14d, 30d]

PR is still WIP, still need to:
- [x] update UI / copy to reflect changes
- [x] make code clearer

### CODE REVIEW

**Comment by cokes518** [comment]
Rolling Earnings > Earnings (UTC Day)

**Comment by danielcolinjames** [comment]
still lots of little improvements that can be made, but it's working:

**Comment by cokes518** [comment]
I love it

**Comment by PeterMain** [comment]
@danielcolinjames is there anyway to get the tooltip balloon to appear above the bars? I find it blocks the rest of the chart quite a lot.  Could be darker (0,0,0,0.8) with lighter text? it blends into the bg quite a lot also

File: components/Widgets/RewardsTrendWidget.js [hunk 1]
```diff
@@ -72,20 +177,94 @@ const RewardsTrendWidget = ({
...
+          backgroundColor: '#464eff',
+          minBarLength: width,
+          barThickness: width * 0.75,
+          borderRadius: width * 0.75 * 2,
+          borderSkipped: false,
+        },
+        {
+          ...(showTarget && {
+            label: 'Network Average',
+            data: averageEarningsData.map((s) => s?.value || 'N/A'),
```
Comment (matthewcarlreetz) [inline components/Widgets/RewardsTrendWidget.js:251]: Might want to consider making `'N/A'` a constant.

**Review by matthewcarlreetz — APPROVED** [review]

File: components/Widgets/RewardsTrendWidget.js [hunk 2]
```diff
@@ -72,20 +177,94 @@ const RewardsTrendWidget = ({
...
+    return {
+      labels: dataToDisplay.map((s) => {
+        return format(
+          new Date(new Date(s.timestamp.slice(0, 10)).setUTCHours(0, 0, 0, 0)),
+          'MMM d',
+        )
+      }),
+      datasets: [
+        {
+          label: 'Earnings',
```
Comment (cokes518) [inline components/Widgets/RewardsTrendWidget.js]:
```suggestion
          label: 'This Hotspotʼs Earnings',
```

File: components/Widgets/RewardsTrendWidget.js [hunk 3]
```diff
@@ -72,20 +177,94 @@ const RewardsTrendWidget = ({
...
+    return {
+      labels: dataToDisplay.map((s) => {
+        return format(
+          new Date(new Date(s.timestamp.slice(0, 10)).setUTCHours(0, 0, 0, 0)),
+          'MMM d',
+        )
+      }),
+      datasets: [
+        {
+          label: 'This Hotspot's Earnings',
```
Comment (cokes518) [inline components/Widgets/RewardsTrendWidget.js]:
```suggestion
          label: 'This Hotspotʼs Earnings',
```

**Review by cokes518 — APPROVED** [review]

**Pull request merged**